ddtrace/tracer/spancontext.go

```diff
@@ -217,21 +217,44 @@ func (t *trace) setTag(key, value string) {
 	t.tags[key] = value
 }
 
-func (t *trace) setSamplingPriorityLocked(service string, p int, sampler samplernames.SamplerName, rate float64) {
+func (t *trace) setPropagatingTag(key, value string) {
+	if t.propagatingTags == nil {
+		t.propagatingTags = make(map[string]string, 1)
+	}
+	t.propagatingTags[key] = value
+}
+
+func (t *trace) setSamplingPriorityLocked(service string, p int, sampler samplernames.SamplerName, rate float64, span *span) {
 	if t.locked {
 		return
 	}
 	if t.priority == nil {
 		t.priority = new(float64)
 	}
 	*t.priority = float64(p)
-	if sampler != samplernames.Upstream {
-		if t.upstreamServices != "" {
-			t.setTag(keyUpstreamServices, t.upstreamServices+";"+compactUpstreamServices(service, p, sampler, rate))
-		} else {
-			t.setTag(keyUpstreamServices, compactUpstreamServices(service, p, sampler, rate))
-		}
+	_, ok := t.propagatingTags[keyDecisionMaker]
+	if p > 0 && !ok {
+		t.setServiceDecisionMaker(service, sampler, span)
+	}
+	if p <= 0 && ok {
+		delete(t.propagatingTags, keyDecisionMaker)
+	}
+}
+
+// setServiceDecisionMaker sets the service and sampler which took the sampling decision made for this trace and span.
+// It is not safe for concurrent use.
+func (t *trace) setServiceDecisionMaker(service string, sampler samplernames.SamplerName, s *span) {
+	serviceHash := Hash(service)
+	tr, haveTracer := internal.GetGlobalTracer().(*tracer)
```

Comment:
```suggestion
	tr, ok := internal.GetGlobalTracer().(*tracer)
```
I think there aren't any other `ok` values in this small function to warrant a special non-standard name.

Also, I suspect, based on the answer to the previous question, that we may be able to simply return if `!ok` or add a default value.

Reply:
You forgot this one.